src/core/apiLayer/productSearchOptionsService.ts

```diff
@@ -0,0 +1,112 @@
+import { stringify } from "query-string";
+
+import { pkbUrl } from "@temp/constants";
+
+export enum OptionType {
+  ChannelTitle = "channelTitle",
+  JobTitle = "jobTitle",
+  JobFunction = "jobFunction",
+  Location = "location",
+}
+
+export interface Option {
+  type: OptionType;
+  label: string;
+  extraLabel?: string;
+  value: string | number;
+}
+
+const fetchJobTitle = async (text: string): Promise<Option[]> => {
+  const query = stringify({ text, limit: 5 });
+  const headers = { "Accept-Language": "en" }; // @todo: Get from user
+  const response = await fetch(`${pkbUrl}job-titles/?${query}`, { headers })
+    .then(response => response.json())
+    .catch(() => []);
+
+  if (!Array.isArray(response?.results)) {
+    return [];
+  }
+
+  return response.results.map(i => {
+    return {
+      type: OptionType.JobTitle,
+      label: i.name,
+      value: i.id,
+    };
+  });
+};
+
+const fetchJobFunctions = async (text: string): Promise<Option[]> => {
+  const query = stringify({ text });
+  const headers = { "Accept-Language": "en" }; // @todo: Get from user
```

Comment:
this seems to repeat itself throughout the frontend - not only pkb, but vacancy service too
can we abstract it somehow?

Reply:
fixed